lib/ui.rb

```diff
@@ -0,0 +1,32 @@
+class UI
+  def initialize(display:, prompter:)
+    @display = display
+    @prompter = prompter
+  end
+
+  def get_move
+    prompter.call
+  end
+
+  def display_board(board)
+    display.present(board)
+  end
+
+  def with_greeting_and_salutation(&block)
+    display.message(welcome_message)
+    block.call
+    display.message(goodbye_message)
+  end
+
+  private
+
+  attr_reader :display, :prompter
+
+  def welcome_message
```

Comment:
I feel that these methods can be omitted and instead just plucking the strings directly

```ruby
def with_greeting_and_salutation(&block)
  display.message('Welcome to Tic Tac Toe!')
  block.call
  display.message('Thank you for playing. Goodbye!')
end
```

As they currently are, the private methods are not being reused, nor are they extracting complexity from the public ones.